File: android/app/src/main/kotlin/com/thisissandipp/one/contacts/models/PhoneNumber.kt
package com.thisissandipp.one.contacts.models

data class PhoneNumber(
    val number: String? = "",
    val normalizedNumber: String? = "",
    val label: String? = "",
    val isPrimary: Boolean? = false,
) {
    fun toMap(): Map<String, Any?> {
        return mapOf(
            "number" to number,
            "normalizedNumber" to normalizedNumber,
            "label" to label,
            "isPrimary" to isPrimary,
        )
    }
}

File: lib/contacts/domain/models/phone_number.dart
import 'package:freezed_annotation/freezed_annotation.dart';
import 'package:one/utils/utils.dart';

part 'phone_number.freezed.dart';
part 'phone_number.g.dart';

@freezed
abstract class PhoneNumber with _$PhoneNumber {
  const factory PhoneNumber({
    @Default('') String number,
    @Default('') String normalizedNumber,
    @Default('') String label,
    @Default(false) bool isPrimary,
  }) = _PhoneNumber;

  factory PhoneNumber.fromJson(JsonMap json) => _$PhoneNumberFromJson(json);
}
